src/router/router.js

```diff
@@ -55,7 +61,20 @@ const isObject = (v) => typeof v === 'object' && v !== null
 
 const isString = (v) => typeof v === 'string'
 
-export const matchHash = (path, routes = []) => {
+const setRouteQueryParams = (routeItem, queryParams = new URLSearchParams()) => {
+  const queryParamsData = {}
+  const queryParamsEntries = [...queryParams.entries()]
+
+  for (let i = 0; i < queryParamsEntries.length; i++) {
+    queryParamsData[queryParamsEntries[i][0]] = queryParamsEntries[i][1]
+  }
+
+  routeItem.queryParams = queryParamsData
+
+  return routeItem
+}
+
+export const matchHash = (path, queryParams = new URLSearchParams(), routes = []) => {
```

Comment:
I would not instantiate always new URLSearchParams() here. That is relatively expensive operation, so I would only do that if we actually have queryParams